### ISSUE TITLE
laguerre_l testsuite failure on i386

### ISSUE THREAD

**Issue by jdemeyer** [fredrik-johansson/arb#79]
On this Linux 32-bit system:
```
Linux arando 3.13.0-40-generic #69-Ubuntu SMP Thu Nov 13 17:56:26 UTC 2014 i686 i686 i686 GNU/Linux
```

testing arb-2.8.0 gives
```
laguerre_l....FAIL: consistency

iter = 444

n = (4 + 0j)  +/-  (0, 0j)

m = (-4 + 0j)  +/-  (0, 0j)

z = (-4503462724239360.00001525855623 - 0.499999999999999999999999996769j)  +/-  (7e-61, 6.91e-77j)

res1 = (1.71385883572304972605522896443e+61 + 7.61129353418828371335823987317e+45j)  +/-  (4, 1.78e-15j)

res2 = (1.5222587068376557286128225538e+46 + 5.07029412715334721112563931956e+30j)  +/-  (1.62e-17, 5.65e-33j)

res3 = (nan + 0j)  +/-  (inf, 0j)

s = (nan + 5.07029412715334721112563931956e+30j)  +/-  (inf, 2.03e-31j)
```

**Comment by fredrik-johansson** [comment]
Hopefully fixed in 4161094ac2fa1d3e65ab06af599edc95ce60628c. Not tested on a 32-bit system, but it was easy enough to produce a similar failure on 64-bit by tweaking the test parameters.

The underlying problem is that a division by zero sometimes only sets the real part or the imaginary part of an acb_t to [nan +/- inf]. This is fine with the understanding that the imaginary part alone is meaningless when the real part contains nan (and vice versa).

However, acb_overlaps() and acb_contains() compare the real and imaginary parts separately without taking this into account, so using them to test identities can give spurious failures.

Propagation and detection of non-finite real and imaginary parts should probably be made more aggressive, to improve overall robustness. But for the time being, I'm content with putting in an extra finiteness check in any test function where this kind of failure occurs.

When examining this bug, I noticed that the recurrence relation tested in t-laguerre_l actually is inconsistent with the implemented definition of the function when n is a negative integer, and this leads to further spurious test failures. I don't know which definition is more useful (this is one of those cases where Mathematica contradicts itself and thus cannot be used as a reference), so I've just changed the function implementation to make this case indeterminate.
